addons/web/tests/test_serving_base.py

```diff
@@ -451,3 +451,148 @@ def test_performance_25000(self):
+        """
+
+        self.assertXMLEqual(contents, expected)
+
+    def test_replace_in_debug_mode3(self):
+        """Text outside of a div which will replace a whole template
+        becomes outside of the template
+        This doesn't mean anything in terms of the business of template inheritance
+        But it is in the XPATH specs"""
+        self.modules = [
+            ('module_1_file_1', None, 'module_1'),
+        ]
+        self.template_files = {
+            'module_1_file_1': b"""
+                <templates id="template" xml:space="preserve">
+                    <form t-name="template_1_1" random-attr="gloria">
+                        <div>At first I was afraid</div>
+                    </form>
+                    <t t-name="template_1_2" t-inherit="template_1_1" t-inherit-mode="extension">
+                        <xpath expr="." position="replace">
+                            <div>
+                                And I grew strong
+                                <p>And I learned how to get along</p>
+                            </div>
+                            And so you're back
+                        </xpath>
+                    </t>
+                </templates>
+                """,
+        }
+
+        contents = HomeStaticTemplateHelpers.get_qweb_templates(addons=self._get_module_names(), debug=True)
+        expected = b"""
+            <templates>
+                <form>
+                    <!-- Modified by template_1_2 from module_1 -->
+                    And I grew strong
+                    <p>And I learned how to get along</p>
+                </form>
+                And so you're back
+            </templates>
+        """
+
+        self.assertXMLEqual(contents, expected)
+
+    def test_replace_in_nodebug_mode1(self):
+        """Comments already in the arch are ignored"""
+        self.modules = [
+            ('module_1_file_1', None, 'module_1'),
+        ]
+        self.template_files = {
+            'module_1_file_1': b"""
+                <templates id="template" xml:space="preserve">
+                    <form t-name="template_1_1" random-attr="gloria">
+                        <div>At first I was afraid</div>
+                    </form>
+                    <t t-name="template_1_2" t-inherit="template_1_1" t-inherit-mode="extension">
+                        <xpath expr="." position="replace">
+                            <div>
+                                <!-- Random Comment -->
+                                And I grew strong
+                                <p>And I learned how to get along</p>
+                                And so you're back
+                            </div>
+                        </xpath>
+                    </t>
+                </templates>
+                """,
+        }
+
+        contents = HomeStaticTemplateHelpers.get_qweb_templates(addons=self._get_module_names(), debug=False)
+        expected = b"""
+            <templates>
```

Comment:
I didn't expect that result. Why should it be considered good?

Reply:
@Polymorphe57 
It is intentional
https://github.com/odoo/odoo/blob/13.0/addons/web/controllers/main.py#L490

Reply:
I don't think it is good. Ok, you are not in debug mode and you shouldn't add the commentaries linked to the xpaths but Is it a good idea to remove all comments?
I mean is is true that "they come from other templates extending..."?

   ```
 def _remove_inheritance_comments(self, inherited_template):
        '''Remove the comments added in the template already, they come from other templates extending
        the base of this inheritance
        :param inherited_template:
        '''
        for comment in inherited_template.xpath('//comment()'):
            if re.match(COMMENT_PATTERN, comment.text.strip()):
                comment.getparent().remove(comment)
```